contentcuration/contentcuration/frontend/channelEdit/vuex/contentNode/mutations.js

```diff
@@ -0,0 +1,114 @@
+import Vue from 'vue';
+import difference from 'lodash/difference';
+import union from 'lodash/union';
+import { contentNodeLastSavedState } from './index';
+
+function mergeContentNode(contentNodesMap, contentNode) {
+  contentNodeLastSavedState.storeLastSavedState(contentNode);
+  return {
+    ...contentNodesMap,
+    [contentNode.id]: {
+      ...contentNodesMap[contentNode.id],
+      ...contentNode,
+    },
+  };
+}
+
+export function ADD_CONTENTNODE(state, contentNode) {
+  state.contentNodesMap = mergeContentNode(state.contentNodesMap, contentNode);
+}
+
+export function ADD_CONTENTNODES(state, contentNodes = []) {
+  state.contentNodesMap = contentNodes.reduce((contentNodesMap, contentNode) => {
+    return mergeContentNode(contentNodesMap, contentNode);
+  }, state.contentNodesMap);
+}
+
+export function REMOVE_CONTENTNODE(state, contentNodeId) {
+  Vue.delete(state.contentNodesMap, contentNodeId);
+}
+
+export function UPDATE_CONTENTNODE(
+  state,
+  { id, title = null, description = null, thumbnailData = null, language = null, license = null, license_description = null, copyright_holder = null, author = null, role_visibility = null, aggregator = null, provider = null, extra_fields = null } = {}
+) {
+  if (!id) {
+    throw ReferenceError('id must be defined to update a contentNode set');
+  }
+  const contentNode = state.contentNodesMap[id];
+  if (title !== null) {
+    contentNode.title = title;
+  }
+  if (description !== null) {
+    contentNode.description = description;
+  }
+  if (
+    thumbnailData !== null &&
+    ['thumbnail', 'thumbnail_url', 'thumbnail_encoding'].every(attr => thumbnailData[attr])
+  ) {
+    contentNode.thumbnail = thumbnailData.thumbnail;
+    contentNode.thumbnail_url = thumbnailData.thumbnail_url;
+    contentNode.thumbnail_encoding = thumbnailData.thumbnail_encoding;
+  }
+  if (language !== null) {
```

Comment:
We might want to allow this either way as nodes without a language set will default to their parent node's language (or the first ancestor to have a language set)

Reply:
To avoid these concerns can create a non-`null` default value as a comparator here.